src/app/common/messagedisplay.ts

```diff
@@ -36,20 +37,38 @@ export abstract class MessageDisplay {
     return this.rows.length;
   }
 
+  selectedRowIds(): number[] {
+    return Object.keys(this.selectedMessageIds).map((msgIndex) => this.findRowByMessageId(parseInt(msgIndex, 10)));
+  }
+
+  // msgIds where the value is true
   selectedMessageIds(): number[] {
-    return Object.keys(this.selectedRowIds).map((rowIndex) => this.getRowMessageId(parseInt(rowIndex, 10)));
+    return Object.keys(this.msgIdsSelected).filter((msgId) => this.msgIdsSelected[parseInt(msgId, 10)]).map((msgId) => parseInt(msgId, 10));
```

Comment:
I don't think these parseInts are necessary: unless we lie in the type definition of the attribute and these isn't actually `{ [number]: boolean }` ;)

Reply:
Object.keys makes things back into strings :(